platform/api/aws/network.go

```diff
@@ -340,3 +348,115 @@ func (a *API) getVPCID(sgId string) (string, error) {
 	}
 	return "", fmt.Errorf("no vpc found for security group %v", sgId)
 }
+
+func (a *API) validateNetworkResources(sg *ec2.SecurityGroup) (bool, error) {
+	if sg.VpcId == nil || *sg.VpcId == "" {
+		return false, nil
+	}
+
+	vpcs, err := a.ec2.DescribeVpcs(&ec2.DescribeVpcsInput{
+		VpcIds: []*string{
+			sg.VpcId,
+		},
+	})
+	if err != nil || len(vpcs.Vpcs) < 1 {
+		return false, nil
+	}
+
+	subnets, err := a.ec2.DescribeSubnets(&ec2.DescribeSubnetsInput{
+		Filters: []*ec2.Filter{
+			{
+				Name: aws.String("vpc-id"),
+				Values: []*string{
+					sg.VpcId,
+				},
+			},
+		},
+	})
+	if err != nil || len(subnets.Subnets) < 1 {
+		// Delete the VPC to remove all networking resources used by kola
+		// as they will be recreated after this check.
+		err = a.DeleteVPC(sg.GroupId, vpcs.Vpcs[0])
+		return false, err
+	}
+
+	return true, nil
+}
+
+func (a *API) DeleteVPC(sgId *string, vpc *ec2.Vpc) error {
+	_, err := a.ec2.DeleteSecurityGroup(&ec2.DeleteSecurityGroupInput{
+		GroupId: sgId,
+	})
+	if err != nil {
+		return fmt.Errorf("deleting security group: %v", err)
```

Comment:
Are the resource IDs included in `err`?  If not, they should be added to these error messages.